integrations/tensorflow/e2e/keras/BUILD

```diff
@@ -372,3 +373,121 @@ iree_py_binary(
         "//integrations/tensorflow/bindings/python/pyiree/tf/support",
     ],
 )
+
+# Keyword Spotting Tests:
+KEYWORD_SPOTTING_MODELS = [
+    "svdf",
+    "svdf_resnet",
+    "ds_cnn",
+    "gru",
+    "lstm",
+    "cnn_stride",
+    "cnn",
+    "tc_resnet",
+    "crnn",
+    "dnn",
+    "att_rnn",
+    "att_mh_rnn",
+    "mobilenet",
+    "mobilenet_v2",
+    "xception",
+    "inception",
+    "inception_resnet",
+    "ds_tc_resnet",
+]
+
+iree_e2e_cartesian_product_test_suite(
+    name = "keyword_spotting_tests",
+    srcs = ["keyword_spotting_streaming_test.py"],
+    failing_configurations = [
+        {
+            # Failing on IREE:
+            "model": [
+                "att_mh_rnn",  # b/147824465
+                "att_rnn",  # b/147824465
+                "crnn",  # b/162067867
+                "ds_tc_resnet",
+                "gru",  # b/162067867
+                "lstm",  # b/162067867
+                "svdf_resnet",  # b/171512071
+                "xception",  # b/171512071
+            ],
+            "target_backends": [
+                "iree_vmla",
+                "iree_llvmjit",
+                "iree_vulkan",
+            ],
+        },
+    ],
+    flags_to_values = {
+        "reference_backend": "tf",
+        "mode": "non_streaming",
+        "model": KEYWORD_SPOTTING_MODELS,
+        "target_backends": [
+            "tf",
+            "tflite",
+            "iree_vmla",
+            "iree_llvmjit",
+            "iree_vulkan",
+        ],
+    },
+    main = "keyword_spotting_streaming_test.py",
+    deps = INTREE_TENSORFLOW_PY_DEPS + NUMPY_DEPS + [
+        "//integrations/tensorflow/bindings/python/pyiree/tf/support",
+        "@kws_streaming//:models_lib",
+        "@kws_streaming//:train_lib",
+    ],
+)
+
+iree_e2e_cartesian_product_test_suite(
+    name = "keyword_spotting_internal_streaming_tests",
```

Comment:
So I know we use this in another place too, but it's a bit odd-sounding to me. "external" tests are "external" in the sense that they're non-hermetic and require external resources. The opposite of that isn't really "internal" which makes me think that we accidentally open-sourced something that shouldn't have been :grin:. How about "hermetic" or just dropping this qualifier entirely. In Bazel, hermetic is the assumption. If you're doing something non-hermetic then you qualify it.

Reply:
This actually refers to the fact that the models are ran in internal streaming mode, meaning that stateful variables are used to track the model's internal state. External mode (which I haven't implemented the test for because it's a nightmare) is when the user has to pass around the model's internal state as an input to the model.

I guess I could rename the test `keyword_spotting_internal_state_streaming_tests`, but someone could just as easily misinterpret that as referring to "state streaming" test that don't rely on internal data? also it's a mouthful 😅

Reply:
Ah. Hmm, but for the cifar tests "internal" vs "external" seems to refer to whether we fetch the weights from the web? https://github.com/google/iree/blob/c8c48cdde515029042f820e1eeb4f124d3a0aa75/integrations/tensorflow/e2e/keras/BUILD#L166. I think that's going to be confusing...

Reply:
We could rename those to be hermetic/non-hermetic?

Reply:
internal/external is the terminology used by `kws_streaming`.

Reply:
Is it used in the same sense for the cifar tests? Can we change that usage?

Reply:
I think that's what I suggested? For clarity I'm suggesting:

```
large_cifar10_tests
enormous_cifar10_tests
cifar10_non_hermetic_tests
imagenet_non_hermetic_tests
keyword_spotting_tests
keyword_spotting_internal_streaming_tests
keyword_spotting_external_streaming_tests
```

Reply:
Whoops I missed your earlier comment! Yes let's merge this for now and if you can rename those other tests in a follow-up, that would be great.

Reply:
Continued in https://github.com/google/iree/pull/3642.